pkgs/sdk-pkg/src/genie/libs/sdk/apis/iosxe/eigrp/verify.py

```diff
@@ -122,3 +122,60 @@ def verify_eigrp_neighbors(
             return set(neighbors).issubset(eigrp_neighbors)
         log.error(f"Please, provide a valid format for AS, Neighbors, ip and/or vrf")
         continue
+
+
+def verify_eigrp_router_id(
+    device,
+    router_id=None,
+    vrf="default",
+    AS_N=None,
+    ip="ipv4",
+    max_time=60,
+    check_interval=10,
+):
+    """verify EIGRP IDs for a given vrf and AS active instance for ipv4 or ipv6
+    Args:
+      device (obj): Device object
+      router_id = None (list): List of router ID to check
+      vrf = "default" (str): Name of the vrf by default set to "default"
+      AS_N = None (int) : Autonomous System
+      ip = "ipv4" (str): Protocol ip set by default to "ipv4" to change to "ipv6"
+      max_time (`int`): Max time, default: 30
+      check_interval (`int`): Check interval, default: 10
+
+    Returns:
+      True
+      False
+    """
+    assert isinstance(AS_N, int), "AS_N must be int"
+    assert isinstance(vrf, str), "vrf must be str"
+    assert isinstance(router_id, list), "router_id must be list of router IDs"
+    assert AS_N != 0, "AS_N must not be 0"
+    assert ip in ["ipv4", "ipv6"], "ip must be ipv4 or ipv6"
+    timeout = Timeout(max_time, check_interval)
```

Comment:
why this timeout needed? comment.

Reply:
Timeout is needed to wait before continue for the parsed data